### PR TITLE
tinysfm: add wrap

### PR DESCRIPTION
**Pull request by stephanlachnit**
for https://github.com/digint/tinyfsm

### CODE REVIEW

**Comment by stephanlachnit** [comment]
```
Found ninja-1.10.2 at C:\ProgramData\Chocolatey\bin\ninja.EXE
WARNING: msvc does not support C++11; attempting best effort; setting the standard to C++14

ERROR: Fatal warnings enabled, aborting

A full log can be found at D:\a\wrapdb\wrapdb\_build\meson-logs\meson-log.txt
```

Proposed solution?

File: subprojects/packagefiles/tinyfsm/meson.build [hunk 1]
```diff
@@ -0,0 +1,56 @@
+project('tinyfsm', 'cpp',
+  version: '0.3.3',
+  default_options: ['cpp_std=c++11'])
+
+tinyfsm_dep = declare_dependency(include_directories: include_directories('include'))
+
+# build examples as compile tests
```
Comment (eli-schwartz) [inline subprojects/packagefiles/tinyfsm/meson.build:7]: If these are `build_by_default: false` then they won't be built in CI. :D I wonder what the solution there is, hmm...

If they can be run successfully as a smoketest, then they could be added as `test()`s and then `ninja test` will build them I guess.

**Comment by neheb** [comment]
C++14

**Review by eli-schwartz — COMMENTED** [review]
The fatal warning would be fixed by https://github.com/mesonbuild/meson/pull/10582

Although you could just set it to c++14 I suppose.

File: subprojects/packagefiles/tinyfsm/meson.build (hunk 1 above)
Comment (stephanlachnit) [reply subprojects/packagefiles/tinyfsm/meson.build:7]: They are interactive, so no. I guess a `build_examples` option (disabled by default) is the best option here...

Comment (stephanlachnit) [reply subprojects/packagefiles/tinyfsm/meson.build:6]: Added a `build_examples` option

**Comment by stephanlachnit** [comment]
@eli-schwartz good to merge now?

**Pull request merged**